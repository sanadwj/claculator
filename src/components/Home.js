import React from 'react';

const Home = () => (
  <>
    <h2 className="home">Welcome to our page!</h2>
    {/* eslint-disable-next-line max-len */}
    <p className="lorem">Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab aperiam architecto aspernatur atque commodi earum exercitationem expedita facere hic incidunt itaque laborum magnam minima molestiae molestias praesentium quas quibusdam recusand.</p>
  </>
);

export default Home;
