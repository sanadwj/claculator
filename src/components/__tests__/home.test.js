import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Home from '../Home';

describe('Home', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Router><Home /></Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Creates a component', () => {
    render(<Router><Home /></Router>);
    const element = screen.getByText('Welcome to our page!');
    expect(element).toBeInTheDocument();
  });
});
