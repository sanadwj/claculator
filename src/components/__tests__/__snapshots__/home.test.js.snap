// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
Array [
  <h2
    className="home"
  >
    Welcome to our page!
  </h2>,
  <p
    className="lorem"
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab aperiam architecto aspernatur atque commodi earum exercitationem expedita facere hic incidunt itaque laborum magnam minima molestiae molestias praesentium quas quibusdam recusand.
  </p>,
]
`;

exports[`Home renders correctly 1`] = `
Array [
  <h2
    className="home"
  >
    Welcome to our page!
  </h2>,
  <p
    className="lorem"
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab aperiam architecto aspernatur atque commodi earum exercitationem expedita facere hic incidunt itaque laborum magnam minima molestiae molestias praesentium quas quibusdam recusand.
  </p>,
]
`;
